arch/risc-v/src/bl602/bl602_netdev.c

```diff
@@ -0,0 +1,2113 @@
+
+    case SIOCGIWENCODEEXT: /* Get encoding token mode */
+      do
+        {
+          struct iwreq *        req = (struct iwreq *)arg;
+          struct iw_encode_ext *ext;
+          wifi_mgmr_t *         mgmr = bl602_netdev_get_wifi_mgmr(priv);
+          int                   length;
+
+          assert(mgmr != NULL);
+
+          ext      = (struct iw_encode_ext *)req->u.encoding.pointer;
+          length   = req->u.encoding.length - sizeof(struct iw_encode_ext);
+          ext->alg = IW_ENCODE_ALG_NONE;
+          ext->key_len = strlen(mgmr->wifi_mgmr_stat_info.psk);
+          if (ext->key_len > length)
+            {
+              return -E2BIG;
+            }
+
+          memcpy(ext->key, mgmr->wifi_mgmr_stat_info.psk, ext->key_len);
+
+          return OK;
+        }
+      while (0);
+      break;
+
+    default:
+      nerr("ERROR: Unrecognized IOCTL command: %d\n", cmd);
+      return -ENOTTY; /* Special return value for this case */
+    }
+
+  return OK;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+extern void wifi_main(int argc, char *argv[]);
+extern void wifi_mgmr_start_background(wifi_conf_t *conf);
+
+uint8_t *bl602_netdev_alloc_txbuf(void)
+{
+  irqstate_t irq = enter_critical_section();
+  int        idx = BIT_FFS(TX_BUFF_BIT_SIZE, &g_tx_buf_indicator);
+  if (idx == 0)
+    {
+      leave_critical_section(irq);
+      nwarn("tx buff alloc failed!\r\n");
+      return NULL;
+    }
+
+  ninfo("tx buff alloc:%d\r\n", idx - 1);
+  BIT_CLR(TX_BUFF_BIT_SIZE, idx - 1, &g_tx_buf_indicator);
+  leave_critical_section(irq);
+
+  return g_tx_buff[idx - 1];
+}
+
+/* NOTIC: The release path of tx buffer DO NOT acquire net lock */
+
+void bl602_netdev_free_txbuf(uint8_t *buf)
+{
+  irqstate_t irq;
+  int        idx;
+
+  idx = (tx_buff_t)buf - g_tx_buff;
+
+  assert(idx < BL602_NET_TXBUFF_NUM && idx >= 0);
+  assert(g_tx_buff[idx] == buf);
+
+  ninfo("tx buff free %d\r\n", idx);
+
+  irq = enter_critical_section();
+  assert(!BIT_ISSET(TX_BUFF_BIT_SIZE, idx, &g_tx_buf_indicator));
+  BIT_SET(TX_BUFF_BIT_SIZE, idx, &g_tx_buf_indicator);
+  leave_critical_section(irq);
+}
+
+/****************************************************************************
+ * Name: bl602_net_initialize
+ *
+ * Description:
+ *   Initialize the Ethernet controller and driver
+ *
+ * Input Parameters:
+ *   intf - In the case where there are multiple EMACs, this value
+ *          identifies which EMAC is to be initialized.
+ *
+ * Returned Value:
+ *   OK on success; Negated errno on failure.
+ *
+ * Assumptions:
+ *   Called early in initialization before multi-tasking is initiated.
+ *
+ ****************************************************************************/
+
+static wifi_conf_t g_conf =
+{
+  .country_code = "CN",
+};
+
+static int wifi_manager_process(int argc, FAR char *argv[])
+{
+#if 0
```

Comment:
Can we add a comment about why this code block is disabled

Reply:
In order to save RAM space, we adapt the firmware code to the LP work queue, which is driven by events. In this way, the stack space of the work queue can be used to run firmware code instead of creating a new process and stack. 
Since these codes are not compatible, I will remove these disabled codes.